### PR TITLE
[Enhancement] support MySQL `time_format` function

### LINKED ISSUE #41951: Integrating StarRocks with Metabase
**TL;DR**: it seems Metabase MariaDB/ MySQL driver tries to call `time_format(TIME, varchar)` as part of the `sync-timezone` routine, but StarRocks doesn't provide such function.

### Steps to reproduce the behavior (Required)
1. Start a Metabase instance (either locally or SaaS).
2. Connect Metabase with StarRocks by following https://docs.starrocks.io/docs/integrations/BI_integrations/Metabase/.
3. Trigger a database sync.

### Expected behavior (Required)
1. See StarRocks tables on Metabase.

### Real behavior (Required)
1. Metabase succeeds on validating the connection.
2. I cannot see any tables as part of the database connection I've just created.
3. By checking Metabase logs, I can see a failure while calling `sync-timezone` routine (it seems to be the cause, but not sure).

### StarRocks version (Required)
3.2.3-a40e2f8
 
### Metabase version
Metabase Cloud (v1.48.7)/ v0.48.7

### Metabase logs
```
[6332a73e-e2d8-4264-ab1f-0bd8a95a0d3f] 2024-02-28T07:39:00-03:00 INFO metabase.task.sync-databases Starting sync task for Database 35.
[6332a73e-e2d8-4264-ab1f-0bd8a95a0d3f] 2024-02-28T07:39:00-03:00 WARN metabase.driver.mysql 

********************************************************************************
WARNING: Metabase only officially supports MySQL 5.7/MariaDB 10.2 and above.
All Metabase features may not work properly when using an unsupported version.
********************************************************************************

[6332a73e-e2d8-4264-ab1f-0bd8a95a0d3f] 2024-02-28T07:39:00-03:00 INFO metabase.sync.util STARTING: Sync metadata for mysql Database 35 'StarRocks'
[6332a73e-e2d8-4264-ab1f-0bd8a95a0d3f] 2024-02-28T07:39:00-03:00 INFO metabase.sync.util STARTING: step 'sync-dbms-version' for mysql Database 35 'StarRocks'
[6332a73e-e2d8-4264-ab1f-0bd8a95a0d3f] 2024-02-28T07:39:00-03:00 INFO metabase.sync.util FINISHED: step 'sync-dbms-version' for mysql Database 35 'StarRocks' (151.5 ms)
[6332a73e-e2d8-4264-ab1f-0bd8a95a0d3f] 2024-02-28T07:39:00-03:00 INFO metabase.sync.util STARTING: step 'sync-timezone' for mysql Database 35 'StarRocks'
[6332a73e-e2d8-4264-ab1f-0bd8a95a0d3f] 2024-02-28T07:39:01-03:00 WARN metabase.sync.util Error running step 'sync-timezone' for mysql Database 35 'StarRocks'
java.sql.SQLTransientConnectionException: (conn=601) Getting analyzing error from line 1, column 73 to line 1, column 173. Detail message: No matching function with signature: time_format(TIME, varchar).
	at org.mariadb.jdbc.internal.util.exceptions.ExceptionFactory.createException(ExceptionFactory.java:79)
	at org.mariadb.jdbc.internal.util.exceptions.ExceptionFactory.create(ExceptionFactory.java:158)
	at org.mariadb.jdbc.MariaDbStatement.executeExceptionEpilogue(MariaDbStatement.java:262)
	at org.mariadb.jdbc.ClientSidePreparedStatement.executeInternal(ClientSidePreparedStatement.java:229)
	at org.mariadb.jdbc.ClientSidePreparedStatement.execute(ClientSidePreparedStatement.java:149)
	at org.mariadb.jdbc.ClientSidePreparedStatement.executeQuery(ClientSidePreparedStatement.java:163)
	at com.mchange.v2.c3p0.impl.NewProxyPreparedStatement.executeQuery(NewProxyPreparedStatement.java:1471)
	at clojure.java.jdbc$execute_query_with_params.invokeStatic(jdbc.clj:1090)
	at clojure.java.jdbc$execute_query_with_params.invoke(jdbc.clj:1084)
	at clojure.java.jdbc$db_query_with_resultset_STAR_.invokeStatic(jdbc.clj:1113)
	at clojure.java.jdbc$db_query_with_resultset_STAR_.invoke(jdbc.clj:1093)
	at clojure.java.jdbc$query.invokeStatic(jdbc.clj:1182)
	at clojure.java.jdbc$query.invoke(jdbc.clj:1144)
	at clojure.java.jdbc$query.invokeStatic(jdbc.clj:1160)
	at clojure.java.jdbc$query.invoke(jdbc.clj:1144)
	at metabase.driver.mysql$fn__106633.invokeStatic(mysql.clj:205)
	at metabase.driver.mysql$fn__106633.invoke(mysql.clj:195)
	at clojure.lang.MultiFn.invoke(MultiFn.java:234)
	at metabase.driver.sql_jdbc$fn__111186.invokeStatic(sql_jdbc.clj:76)
	at metabase.driver.sql_jdbc$fn__111186.invoke(sql_jdbc.clj:71)
	at clojure.lang.MultiFn.invoke(MultiFn.java:234)
	at metabase.sync.sync_metadata.sync_timezone$sync_timezone_BANG_.invokeStatic(sync_timezone.clj:41)
	at metabase.sync.sync_metadata.sync_timezone$sync_timezone_BANG_.invoke(sync_timezone.clj:36)
	at clojure.lang.AFn.applyToHelper(AFn.java:154)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.core$apply.invokeStatic(core.clj:669)
	at clojure.core$apply.invoke(core.clj:662)
	at metabase.sync.util$run_step_with_metadata$fn__57032.doInvoke(util.clj:425)
	at clojure.lang.RestFn.invoke(RestFn.java:397)
	at metabase.sync.util$with_start_and_finish_logging_STAR_.invokeStatic(util.clj:132)
	at metabase.sync.util$with_start_and_finish_logging_STAR_.invoke(util.clj:126)
	at metabase.sync.util$with_start_and_finish_debug_logging.invokeStatic(util.clj:150)
	at metabase.sync.util$with_start_and_finish_debug_logging.invoke(util.clj:146)
	at metabase.sync.util$run_step_with_metadata.invokeStatic(util.clj:420)
	at metabase.sync.util$run_step_with_metadata.invoke(util.clj:415)
	at metabase.sync.util$run_sync_operation$fn__57111.invoke(util.clj:531)
	at metabase.sync.util$run_sync_operation.invokeStatic(util.clj:529)
	at metabase.sync.util$run_sync_operation.invoke(util.clj:523)
	at metabase.sync.sync_metadata$sync_db_metadata_BANG_$fn__81408.invoke(sync_metadata.clj:63)
	at metabase.sync.util$do_with_error_handling.invokeStatic(util.clj:191)
	at metabase.sync.util$do_with_error_handling.invoke(util.clj:184)
	at clojure.core$partial$fn__5910.invoke(core.clj:2647)
	at metabase.driver$fn__48779.invokeStatic(driver.clj:725)
	at metabase.driver$fn__48779.invoke(driver.clj:725)
	at clojure.lang.MultiFn.invoke(MultiFn.java:239)
	at metabase.sync.util$sync_in_context$fn__56973.invoke(util.clj:167)
	at metabase.sync.util$with_db_logging_disabled$fn__56970.invoke(util.clj:159)
	at metabase.sync.util$with_start_and_finish_logging_STAR_.invokeStatic(util.clj:132)
	at metabase.sync.util$with_start_and_finish_logging_STAR_.invoke(util.clj:126)
	at metabase.sync.util$with_start_and_finish_logging$fn__56957.invoke(util.clj:144)
	at metabase.sync.util$with_sync_events$fn__56952.invoke(util.clj:118)
	at metabase.sync.util$with_duplicate_ops_prevented$fn__56939.invoke(util.clj:90)
	at metabase.sync.util$do_sync_operation.invokeStatic(util.clj:216)
	at metabase.sync.util$do_sync_operation.invoke(util.clj:210)
	at metabase.sync.sync_metadata$sync_db_metadata_BANG_.invokeStatic(sync_metadata.clj:62)
	at metabase.sync.sync_metadata$sync_db_metadata_BANG_.invoke(sync_metadata.clj:58)
	at metabase.task.sync_databases$sync_and_analyze_database_STAR__BANG_.invokeStatic(sync_databases.clj:94)
	at metabase.task.sync_databases$sync_and_analyze_database_STAR__BANG_.invoke(sync_databases.clj:77)
	at metabase.task.sync_databases$sync_and_analyze_database_BANG_.invokeStatic(sync_databases.clj:113)
	at metabase.task.sync_databases$sync_and_analyze_database_BANG_.invoke(sync_databases.clj:101)
	at metabase.task.sync_databases.SyncAndAnalyzeDatabase.execute(sync_databases.clj:118)
	at org.quartz.core.JobRunShell.run(JobRunShell.java:202)
	at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:573)
Caused by: org.mariadb.jdbc.internal.util.exceptions.MariaDbSqlException: Getting analyzing error from line 1, column 73 to line 1, column 173. Detail message: No matching function with signature: time_format(TIME, varchar).
	at org.mariadb.jdbc.internal.util.exceptions.MariaDbSqlException.of(MariaDbSqlException.java:34)
	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.exceptionWithQuery(AbstractQueryProtocol.java:195)
	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.exceptionWithQuery(AbstractQueryProtocol.java:178)
	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.executeQuery(AbstractQueryProtocol.java:322)
	at org.mariadb.jdbc.ClientSidePreparedStatement.executeInternal(ClientSidePreparedStatement.java:220)
	... 59 more
Caused by: java.sql.SQLException: Getting analyzing error from line 1, column 73 to line 1, column 173. Detail message: No matching function with signature: time_format(TIME, varchar).
	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.readErrorPacket(AbstractQueryProtocol.java:1693)
	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.readPacket(AbstractQueryProtocol.java:1555)
	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.getResult(AbstractQueryProtocol.java:1518)
	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.executeQuery(AbstractQueryProtocol.java:319)
	... 60 more
```

### PR DESCRIPTION
**Pull request by dorsegal**
## Why I'm doing:
According to Metabase format_time is a must function.
https://github.com/metabase/metabase/issues/50930#issuecomment-2520402223
https://github.com/metabase/metabase/issues/28081
## What I'm doing:
Adding format_time function like MySQL

Fixes https://github.com/StarRocks/starrocks/issues/41951

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.4
  - [x] 3.3
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0

### CODE REVIEW

File: be/src/exprs/time_functions.cpp [hunk 1]
```diff
@@ -3222,4 +3222,85 @@ Status TimeFunctions::last_day_close(FunctionContext* context, FunctionContext::
...
+    if (columns.size() != 2) {
+        return Status::InvalidArgument("FORMAT_TIME requires exactly 2 arguments");
+    }
+
+    const auto& time_column = columns[0];
+    const auto& format_column = columns[1];
+
+    if (time_column->is_nullable() || format_column->is_nullable()) {
+        return Status::InvalidArgument("FORMAT_TIME arguments cannot be NULL");
+    }
```
Comment (stdpain) [inline be/src/exprs/time_functions.cpp:3236]:
```suggestion
        RETURN_IF_COLUMNS_ONLY_NULL(columns);
```

File: be/src/exprs/time_functions.cpp [hunk 2]
```diff
@@ -3222,4 +3222,85 @@ Status TimeFunctions::last_day_close(FunctionContext* context, FunctionContext::
...
+
+    auto* time_col = ColumnHelper::cast_to<TYPE_TIME>(time_column);
+    auto* format_col = ColumnHelper::cast_to<TYPE_VARCHAR>(format_column);
+
```
Comment (stdpain) [inline be/src/exprs/time_functions.cpp:3240]: using 
```
    auto time_viewer = ColumnViewer<TYPE_TIME>(time_column);
    auto format_viewer = ColumnViewer<TYPE_VARCHAR>(format_column);
```

File: be/src/exprs/time_functions.cpp [hunk 3]
```diff
@@ -3222,4 +3222,83 @@ Status TimeFunctions::last_day_close(FunctionContext* context, FunctionContext::
...
+    
+    auto time_viewer = ColumnViewer<TYPE_TIME>(time_column);
+    auto format_viewer = ColumnViewer<TYPE_VARCHAR>(format_column);
+
+    const size_t size = time_column->size();
+    auto builder = ColumnHelper::get_builder<TYPE_VARCHAR>(size);
+
+    for (size_t i = 0; i < size; ++i) {
+        TimeValue time_val = time_viewer.value(i);
+        std::string_view format_str = format_viewer.value(i);
```
Comment (stdpain) [inline be/src/exprs/time_functions.cpp:3244]: if (time_viewer.is_null(i) || format_viewer.is_null(i) {
    builder->append_null();
    continue;
}

**Review by stdpain — DISMISSED** [review]

**Review by stdpain — DISMISSED** [review]

**Comment by stdpain** [comment]
example:
```
mysql> SELECT @@GLOBAL.time_zone AS global_tz, 
    ->        @@system_time_zone AS system_tz,
    ->        time_format(  
    ->           timediff(
    ->             now(), convert_tz(now(), @@GLOBAL.time_zone, '+00:00')
    ->           ),
    ->          '%H:%i' ) AS 'offset';
+---------------+---------------+--------+
| global_tz     | system_tz     | offset |
+---------------+---------------+--------+
| Asia/Shanghai | Asia/Shanghai | 00:00  |
+---------------+---------------+--------+
1 row in set (0.01 sec)
```

**Review by stdpain — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Comment by dorsegal** [comment]
Awesome! Thank you. Will it it on 3.3.7?

**Comment by stdpain** [comment]
> Awesome! Thank you. Will it it on 3.3.7?

will be release in 3.3.8

**Pull request merged**